lib/rex/socket/ssl_tcp.rb

```diff
@@ -72,13 +72,18 @@ def initsock(params = nil)
 		self.sslctx  = OpenSSL::SSL::SSLContext.new(version)
 
 		# Configure the SSL context
-		# TODO: Allow the user to specify the verify mode and callback
+		# TODO: Allow the user to specify the verify mode callback
 		# Valid modes:
 		#  VERIFY_CLIENT_ONCE
 		#  VERIFY_FAIL_IF_NO_PEER_CERT
 		#  VERIFY_NONE
 		#  VERIFY_PEER
-		self.sslctx.verify_mode = OpenSSL::SSL::VERIFY_PEER
+		if params.ssl_verify_mode
+			self.sslctx.verify_mode = OpenSSL::SSL.const_get("VERIFY_#{params.ssl_verify_mode}".intern)
```

Comment:
This isn't a good fit, also, the existing callback is used to save the context, and calling this callback would break it. How about using a Proc and calling it after the existing handler?